core/config/engine.cpp

```diff
@@ -210,6 +210,28 @@ void Engine::get_singletons(List<Singleton> *p_singletons) {
 	}
 }
 
+void Engine::set_buffered_logger(BufferedSingletonLogger *logger) {
+	buffered_log = logger;
+}
+
+Array Engine::get_log_buffer() {
+	if (GLOBAL_GET("logging/buffered_logging/enable_buffered_logging")) {
+		return buffered_log->flushMessageBuffer();
+	} else {
+		ERR_PRINT(vformat("project setting 'logging/buffered_logging/enable_buffered_logging' must be true in order to get the buffered log"));
+		return Array();
+	}
```

Comment:
Use Godot's [error macros](https://docs.godotengine.org/en/latest/development/cpp/common_engine_methods_and_macros.html#error-macros) for more concise code:

```suggestion
	ERR_FAIL_COND_V_MSG(!GLOBAL_GET("logging/buffered_logging/enable_buffered_logging"), Array(), "The project setting \"logging/buffered_logging/enable_buffered_logging\" must be `true` to get the buffered output log.");

	return buffered_log->flushMessageBuffer();
```